examples/synchronization-mutex-tmpl-level-legacy.yaml

```diff
@@ -28,18 +27,22 @@ spec:
 
   - name: acquire-lock
     synchronization:
-      mutex:
+      mutex: # deprecated: v3.5 and before
         name: welcome
+      # mutexes: # v3.6 and after
+      #   - name: welcome
```

Comment:
I'm inclined to just move these to internal tests and otherwise remove the legacy examples entirely from the user-facing `examples/` dir

and flipped syntax + in-line comments here is a stopgap, especially as the in-line deprecation is hard to understand without the future syntax right next to it (and users are likely to miss `-legacy` name too)